### ISSUE TITLE
We should transpile tachyons-custom to expose more variables

### ISSUE THREAD

**Issue by johnotander** [tachyons-css/tachyons-sass#26]

**Comment by ptrikutam** [comment]
First off, thank you for all your hard work -- this library kicks ass as does the entire tachyons ecosystem.

I've started to take a swing at this task, you can see the current WIP at https://github.com/ptrikutam/tachyons-sass/tree/tachyons-custom

Comparison link for reference: https://github.com/tachyons-css/tachyons-sass/compare/master...ptrikutam:tachyons-custom?expand=1

Changes I've made: 

- Update the package.json to pull from `tachyons-custom`
- Update the `build.js` script to:
    - Update `tachyons` to `tachyons-custom` wherever appropriate
    - remove the conditions on line 17 that skip over `_colors` and `_media-queries` since those are both contained within `_variables`
    - skip over `_variables` in the transpiling process because I was getting weird behavior for [these lines](https://github.com/tachyons-css/tachyons-custom/blob/master/src/_variables.css#L13-L21).
    - Update the regex to add a space before the `Modules` comment matcher since that's how it is in the `tachyons-custom` repo.
- Manually update the partly transpiled `_variables.scss` to fix weirdly generated code from [these lines](https://github.com/tachyons-css/tachyons-custom/blob/master/src/_variables.css#L13-L21) (see below).

I'm a bit stuck on:

- [This line](https://github.com/ptrikutam/tachyons-sass/blob/tachyons-custom/scss/_nested.scss#L45) in the auto-generated `_nested.scss` is a bit screwed up (pulling from [this line](https://github.com/tachyons-css/tachyons-custom/blob/master/src/_nested.css#L38) in the original `tachyons-custom` repo).
- As mentioned above, [these lines](https://github.com/tachyons-css/tachyons-custom/blob/master/src/_variables.css#L13-L21) generate the following code in `_variables.scss` when I auto-transpile them:

```
$sans-serif: -apple-system, BlinkMacSystemFont,                'avenir next', avenir,                helvetica, 'helvetica neue',                ubuntu,                roboto,                noto,                'segoe ui',                arial,                sans-serif !default;
,
,
,
,
,
,
,
```

Should I just manually go in and change these files? Or is this an issue with the `cssScss` library?

I'm a bit new to this repo and the libraries in use, so sorry if this is asking a simple question or if I'm making more work for you.

**Comment by johnotander** [comment]
Thanks for spiking this out! Looks like you've encountered a few issues downstream : )

Firstly, the `tachyons-custom` problem in `_nested` appears to result from how we're handling a variable that doesn't have an `em` unit (line height doesn't have a unit value). I think how we will actually want to handle this is to create a separate variable (in `tachyons-custom`) to keep things the cleanest which is then referenced. That way it will all transpile cleanly.

As far as `_variables` is concerned with the font stack, it looks like there's something wonky going on in [`css-scss`](https://github.com/jxnblk/css-scss). I'll try and dig into this a bit and see if I can get that patched.

I'd love to keep any need out for custom modifications after the transpilation process. So we'll need to work out these issues before merging in, but I think we can get there 💟

**Comment by ptrikutam** [comment]
> Firstly, the tachyons-custom problem in _nested appears to result from how we're handling a variable that doesn't have an em unit (line height doesn't have a unit value). I think how we will actually want to handle this is to create a separate variable (in tachyons-custom) to keep things the cleanest which is then referenced. That way it will all transpile cleanly.

Ah, that makes sense. It's getting late here but maybe I'll try to submit a PR on the `tachyons-custom` repo tomorrow if I get a chance!

> As far as _variables is concerned with the font stack, it looks like there's something wonky going on in css-scss. I'll try and dig into this a bit and see if I can get that patched.

If I was to venture a completely uneducated guess, it seems to mess up with multi-line variables. The `serif`font stack seems to transpile just fine.

> I'd love to keep any need out for custom modifications after the transpilation process. So we'll need to work out these issues before merging in, but I think we can get there 💟

Agreed, thanks for weighing in. 

`_colors` and `_media-queries` are currently omitted from the transpiling process. What was the reason for that?

**Comment by johnotander** [comment]
Yeah, if you wanted to open up a pr when you get a chance on `tachyons-custom` when you can that'd be great 👌 

> If I was to venture a completely uneducated guess, it seems to mess up with multi-line variables. The seriffont stack seems to transpile just fine.

Yep, seems to be exactly right. Since that's something going on in `postcss` (traced it down to the parsing in `css-scss`) internals we should prolly just make `tachyons-custom` declare the variable in a single line to avoid that can of worms for now 😬 .

> _colors and _media-queries are currently omitted from the transpiling process. What was the reason for that?

Tbh, I can't remember why. Prolly shouldn't be. Especially since those will change their values in v5 : )

**Comment by ptrikutam** [comment]
> Tbh, I can't remember why. Prolly shouldn't be. Especially since those will change their values in v5 : )

No worries. I've removed those files entirely and they are not being omitted from the transpiling process. I'll update my branch and submit a PR once we get https://github.com/tachyons-css/tachyons-custom/pull/10 merged.

**Comment by johnotander** [comment]
Merged and published as `4.9.1` ❤️
